### PR TITLE
Fix the pause module for python3

### LINKED ISSUE #26278: Pause prompt broken in python3
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
pause

##### ANSIBLE VERSION

```
ansible 2.3.2.0 (detached HEAD 7424e1c417) last updated 2017/06/30 11:12:33 (GMT +200)
  config file = 
  configured module search path = Default w/o overrides
  python version = 3.6.1 (default, Apr  4 2017, 09:40:21) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]

```

##### CONFIGURATION
Default

##### OS / ENVIRONMENT
MacOS

##### SUMMARY
Using the `prompt` functionality of the `pause` module, the interactive input does not stop anymore when pressing Enter. If the input is interrupted using Ctrl+C and the process is continued by pressing "C", the returned `user_input` also contains the newline characters.

##### STEPS TO REPRODUCE

```yaml
---

- name: Input
  pause:
    prompt: "Username?"
  register: r_prompt

- debug:
    msg: "{{r_prompt.user_input}}"
```

- Run this tasks and enter `123<ENTER>456`
- Because process does not continue, press CTRL+C and then C

##### EXPECTED RESULTS
 - expect process to continue when pressing <ENTER>
 - expect `user_input` to be `123`

##### ACTUAL RESULTS

```
TASK [basis : Input] ************************************************************************************************************************************************************************************************************************************************
task path: /Users/phsc/Projects/sandbox/test/roles/basis/tasks/test.yml:3
[basis : Input]
Username?:
Press 'C' to continue the play or 'A' to abort 
ok: [localhost] => {
    "changed": false,
    "delta": 5,
    "rc": 0,
    "start": "2017-06-30 11:21:19.867562",
    "stderr": "",
    "stdout": "Paused for 0.1 minutes",
    "stop": "2017-06-30 11:21:25.623051",
    "user_input": "123\n456"
}

TASK [basis : debug] ************************************************************************************************************************************************************************************************************************************************
task path: /Users/phsc/Projects/sandbox/test/roles/basis/tasks/test.yml:8
ok: [localhost] => {
    "msg": "123\n456"
}
META: ran handlers
META: ran handlers

```

### PR DESCRIPTION
**Pull request by abadger**
* On python3, stdin goes through a buffer that translates from raw bytes
  to text.  this interferes with pause as it (1) performs universal
  newline conversion and therefore '\r' is turned into '\n' and (2) the
  buffering prevents us from getting the typed characters immediately
  (possibly a python3 bug?)  Using the raw byte stream that's behind the
  text decoder fixes these problems.

Unrelated cleanups:
* Use to_text instead of str for conversion into strings to avoid possible tracebacks
* Use either \r or \n as the end of a line.

Fixes #26278
Resolves #26446

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/plugins/action/pause.py

##### ANSIBLE VERSION

```
devel 2.3
```

### CODE REVIEW

**Comment by abadger** [comment]
Test failure on suse seems unrelated (package repository problem). freebsd test is still running.

File: lib/ansible/plugins/action/pause.py [hunk 1]
```diff
@@ -102,8 +104,8 @@ def run(self, tmp=None, task_vars=None):
         # Begin the hard work!
 
         start = time.time()
-        result['start'] = str(datetime.datetime.now())
-        result['user_input'] = ''
+        result['start'] = to_text(datetime.datetime.now())
+        result['user_input'] = b''
```
Comment (alikins) [inline lib/ansible/plugins/action/pause.py:108]: ActionModule.run() results only need to be pickle'able? Or do they need to be JSON serializable?  The plain[1] py3 json.dumps() doesn't understand bytestrings, but not sure if that is an issue here.

Comment (abadger) [reply lib/ansible/plugins/action/pause.py:108]: This is converted to a text string on line 192

Comment (bcoca) [reply lib/ansible/plugins/action/pause.py:108]: i should need neither, only modules should need actual serialization

**Comment by abadger** [comment]
Merged to devel and to the temp-staging-post2.3.2 branch.  Should be in 2.3.3 if we make a 2.3.3 release, otherwise, 2.4.0

**Pull request merged**